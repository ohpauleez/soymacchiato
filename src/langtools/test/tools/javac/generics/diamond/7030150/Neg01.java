/*
 * @test /nodynamiccopyright/
 * @bug 7030150
 * @summary Type inference for generic instance creation failed for formal type parameter
 *          check that explicit type-argument that causes resolution failure is rejected
 * @compile/fail/ref=Neg01.out -XDrawDiagnostics Neg01.java
 */

class Neg01 {

    static class Foo<X> {
        <T> Foo(T t) {}
    }

    Foo<Integer> fi1 = new <String> Foo<>(1);
    Foo<Integer> fi2 = new <String> Foo<Integer>(1);
}
