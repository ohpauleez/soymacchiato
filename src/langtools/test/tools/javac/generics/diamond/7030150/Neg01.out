Neg01.java:15:24: compiler.err.cant.apply.diamond.1: (compiler.misc.diamond: Neg01.Foo), (compiler.misc.infer.no.conforming.assignment.exists: X, int, java.lang.String)
Neg01.java:16:24: compiler.err.cant.apply.symbol.1: kindname.constructor, Foo, T, int, kindname.class, Neg01.Foo<X>, (compiler.misc.no.conforming.assignment.exists: int, java.lang.String)
2 errors
